postgres/infoschema.go

```diff
@@ -415,6 +420,65 @@ func getForeignKeys(conv *internal.Conv, db *sql.DB, table schemaAndName) (forei
 	return foreignKeys, nil
 }
 
+// getIndexes return a list of all indexes for the specified table.
+func getIndexes(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]schema.Index, error) {
+	q := `SELECT
+			irel.relname AS index_name,
```

Comment:
Wow!
I've been starting at this for a while now, and playing with it in psql.
It looks solid, but it's much more complex that I'd expected.

Any insights you can offer for how you constructed this?
I'm wondering if we should add a comment to help with understanding/maintaining this.

Reply:
Yeah, i took major help from our DBA team and some from this stackoverflow answer: https://stackoverflow.com/questions/6777456/list-all-index-names-column-names-and-its-table-name-of-a-postgresql-database/44460269#44460269

Reply:
How about adding a comment:

// Extracting index definitions from PostgreSQL information schema tables is complex.
// See https://stackoverflow.com/questions/6777456/list-all-index-names-column-names-and-its-table-name-of-a-postgresql-database/44460269#44460269
// for background.

Reply:
added comment